src/box/lua/func_adapter.c

```diff
@@ -0,0 +1,280 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2023, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#include "box/lua/func_adapter.h"
+#include "box/lua/tuple.h"
+#include "box/tuple.h"
+#include "core/func_adapter.h"
+#include "lua/utils.h"
+
+/**
+ * Context for lua_func_adapter.
+ */
+struct lua_func_adapter_ctx {
+	/**
+	 * Lua state which stores arguments and is used to call Lua function.
+	 */
+	struct lua_State *L;
+	/**
+	 * A reference to Lua state in Lua registry.
+	 */
+	int coro_ref;
+	/**
+	 * Saved top of L.
+	 */
+	int top_svp;
+	/**
+	 * Index of an element that will be popped next. It is used not to
+	 * remove elements from the middle of the Lua stack.
+	 */
+	int idx;
+};
+
+static_assert(sizeof(struct lua_func_adapter_ctx) <= sizeof(struct func_adapter_ctx),
+	      "sizeof(lua_func_adapter_ctx) must be <= "
+	      "sizeof(func_adapter_ctx)");
+
+/**
+ * Specialization of func_adapter for Lua functions.
+ */
+struct lua_func_adapter {
```

Comment:
`lua_` is the Lua API namespace. We shouldn't use it for Tarantool internals. Probably better call this `func_adapter_lua` (similarly to `port_lua`, `func_lua`).

Reply:
Done. Note that the unit test is still called `lua_func_adapter.c` in order to group all unit tests for Lua.